tests/pixeldataset/io/test_datastores.py

```diff
@@ -0,0 +1,237 @@
+"""Tests for the pixeldataset.io.datastores module.
+
+Copyright (c) 2024 Pixelgen Technologies AB.
+"""
+
+from unittest.mock import patch
+from anndata import AnnData
+from pandas.core.frame import DataFrame
+from pathlib import Path
+
+import pytest
+from pandas.testing import assert_frame_equal
+from pixelator.pixeldataset import PixelDataset
+
+from pixelator.pixeldataset.io.datastores import (
+    FileFormatNotRecognizedError,
+    PixelDataStore,
+    ZipBasedPixelFile,
+    ZipBasedPixelFileWithCSV,
+    ZipBasedPixelFileWithParquet,
+)
+
+
+class TestPixelDataStore:
+    def test_pixel_data_store_guess_from_path_parquet(self, pixel_dataset_file: Path):
+        res = PixelDataStore.guess_datastore_from_path(pixel_dataset_file)
+        assert isinstance(res, ZipBasedPixelFileWithParquet)
+
+    def test_pixel_data_store_guess_from_path_csv(
+        self,
+        setup_basic_pixel_dataset: tuple[
+            PixelDataset, DataFrame, AnnData, dict[str, int], DataFrame, DataFrame
+        ],
+        tmp_path: Path,
+    ):
+        dataset, *_ = setup_basic_pixel_dataset
+        file_target = tmp_path / "dataset.pxl"
+        dataset.save(str(file_target), file_format="csv")
+        res = PixelDataStore.guess_datastore_from_path(file_target)
+        assert isinstance(res, ZipBasedPixelFileWithCSV)
+
+    def test_pixel_data_store_from_file_provides_correct_datastore_parquet(
+        self,
+        setup_basic_pixel_dataset: tuple[
+            PixelDataset, DataFrame, AnnData, dict[str, int], DataFrame, DataFrame
+        ],
+        tmp_path: Path,
+    ):
+        dataset, *_ = setup_basic_pixel_dataset
+        file_target = tmp_path / "dataset.pxl"
+        dataset.save(str(file_target), file_format="parquet")
+        res = PixelDataStore.from_path(file_target)
+        assert isinstance(res, ZipBasedPixelFileWithParquet)
+
+    def test_pixel_data_store_from_file_provides_correct_datastore_csv(
+        self,
+        setup_basic_pixel_dataset: tuple[
+            PixelDataset, DataFrame, AnnData, dict[str, int], DataFrame, DataFrame
```

Comment:
Triplicate DataFrame here?

```suggestion
            PixelDataset, DataFrame, AnnData, dict[str, int]
```

Reply:
I actually think this is correct, since it will return a tuple looking like this:
``` 
dataset,
edgelist,
adata,
metadata,
polarization_scores,
colocalization_scores,
```
Where dataset comes in trice, first for the edgelist, then for the polarization and colocalization scores.